snap/info.go

```diff
@@ -766,10 +766,13 @@ func infoFromSnapYamlWithSideInfo(meta []byte, si *SideInfo) (*Info, error) {
 type NotFoundError struct {
 	Snap     string
 	Revision Revision
+	// Path encodes the path that triggered the not-found error.
+	// It may refer to a file inside the snap or to the snap file itself.
+	Path string
 }
 
 func (e NotFoundError) Error() string {
-	return fmt.Sprintf("cannot find installed snap %q at revision %s", e.Snap, e.Revision)
+	return fmt.Sprintf("cannot find installed snap %q at revision %s (missing file: %q)", e.Snap, e.Revision, e.Path)
```

Comment:
I think the bit about path should be optional, it doesn't seem we use NotFoundError in other random but we might at some point